core/dbt/artifacts/resources/v1/metric.py

```diff
@@ -92,6 +92,12 @@ class MetricTypeParams(dbtClassMixin):
     metrics: Optional[List[MetricInput]] = None
     conversion_type_params: Optional[ConversionTypeParams] = None
 
+    def add_input_measure(self, input_measure) -> None:
```

Comment:
nit: missing type annotation on input_measure

Reply:
also, there's no 'node' equivalent for a MetricTypeParams, but I'm still not sure we want to have this type of mutating method implemented on a resource. 

It's more verbose, but what do you think about implementing this as either `add_input_measure_to_type_params` or `add_input_measure` (since i think its unambiguous where the input measures are) on the `Metric` node in contracts.graph.nodes directly?

Reply:
I can move it to the Metric node.

Reply:
To push this farther, do we need a mutating method? An alternative would be to just use a `set` in `_process_metric_node` for constructing the list, and then at the end of the function convert via `type_params.input_measures=list(measures_set)`. I guess the drawback would be if anywhere else in the code base wanted to add additional measures to the list after the fact, but I'm not sure that's even a workflow we'd want to encourage or support.

Reply:
I looked at this option, but it doesn't fit into that big if/elif/elif/elif/else very well.  The "add_input_measure" method is similar to other methods that we use such as "add_node", and it's usable in other places, should they occur. If you wanted to change the definition of "input_measures" to be a set...